src/transformers/file_utils.py

```diff
@@ -456,6 +456,11 @@ def get_from_cache(
     lock_path = cache_path + ".lock"
     with FileLock(lock_path):
 
+        # If the download just completed while the lock was activated.
+        if os.path.exists(cache_path) and not force_download:
+            # Even if returning early like here, the lock will be released.
+            return cache_path
+
```

Comment:
@aaugustin I'm guessing this is an invariant of the language that the context manager's `__exit__` method is called even when returning early?

Reply:
(Also tagging you as you implemented this in the first place)

The lock was preventing two downloads to write to the same file at the same time. But all processes would still end up downloading the file. With this change, the non-first ones should return the cached file. cc @thomwolf

Reply:
Yes, `__exit__` is always called when you exit the block, regardless of how you exit it.

Reply:
Instead of duplicating these two lines (before taking the lock and after taking the lock), I'd just move them inside the locked block.

It won't change anything in practice but it will be more readable.